examples/http-request-interceptors/http_request_interceptors.bal

```diff
@@ -0,0 +1,65 @@
+import ballerina/http;
+
+// Header names to be set to the request in the request interceptor.
+final string interceptor_header1 = "requestHeader1";
+final string interceptor_header2 = "requestHeader2";
+
+// Header values to be set to the request in the request interceptor.
+final string interceptor_header_value1 = "RequestInterceptor1";
+final string interceptor_header_value2 = "RequestInterceptor2";
+
+// A `Requestinterceptorservice` class implementation. It intercepts the request
+// and adds a header before it is dispatched to the target service. A `RequestInterceptorService`
+// class can have only one resource function.
+service class RequestInterceptor1 {
+    *http:RequestInterceptor;
+
+    // A default resource function, which will be executed for all the requests. 
+    // A `RequestContext` is used to share data between the interceptors.
+    resource function 'default [string... path](http:RequestContext ctx, 
+                        http:Request req) returns http:NextService|error? {
+        // Sets a header to the request inside the interceptor service.
+        req.setHeader(interceptor_header1, interceptor_header_value1);
+        // Returns the next interceptor or the target service in the pipeline. 
+        // An error is returned when the call fails.
+        return ctx.next();
+    }
+}
+
+// An `Interceptor` service class with a specific path. This interceptor can only be 
+// engaged at the service level.
+service class RequestInterceptor2 {
```

Comment:
IMO one interceptor is enough. We can explain this filtering behaviour by improving the comments of `RequestInterceptor1`
Say that `By specifying the accessor and path in the interceptor resource, requests can be filtered to intercept only the matched ones` or something similar to that?

Reply:
+1